cinder/volume/manager.py

```diff
@@ -977,8 +987,10 @@ def delete_volume(self,
             else:
                 self.driver.delete_volume(volume)
         except exception.VolumeIsBusy:
-            LOG.error("Unable to delete busy volume.",
-                      resource=volume)
+            trace_utils.trace(
+                context, trace_utils.ACTION_VOLUME_DELETE,
+                volume, "Unable to delete busy volume.", failed=True
+            )
```

Comment:
Probably not a problem for us, because we run everything with DEBUG anyways, but if you'd want to upstream it, people might just want to enable debug/info messages on certain modules to not overwhelm their logs.

Imho, it must be made clear for them, that enabling debug-logging in a specific module does nothing with regards to the `trace_utils` logs and that it's "all or nothing" for these, because they are only controlled via the `cinder.trace_utils` module's logger. Taking Nova as example, we don't have DEBUG logging everywhere. Default is even INFO.
https://github.com/sapcc/helm-charts/blob/3e771554d3d65d93c2b5ffaf21dc6d42253ac8b0/openstack/nova/values.yaml#L846-L857
But we do enable DEBUG logs for `nova.scheduler`, just not for `nova.scheduler.host_manager`. This kind of control is lost, when the `trace_utils` always use the single logger from `cinder.trace_utils`.

This might be fine if you say "trace_utils is a mandatory feature in Cinder going forward" - but it should be stated explicitly when bringing this upstream. If it's not meant to be mandatory, I'd suggest to pass the module's loggers into the `trace_utils.trace()` calls.

Reply:
I think this needs further consideration, either by restoring the removed log messages or a more clear messaging & separation around the orthogonality of log levels and action tracking.